frontend/src/lib/hooks/useWhyDidIRender.ts

```diff
@@ -2,6 +2,15 @@ import { useMemo, useRef } from 'react'
 
 import { useFeatureFlag } from './useFeatureFlag'
 
+// NOTE: Useful for debugging reasons, you can keep this in a component and enable the feature flag in production
+// to see debugging logs in a production instance and understand why a component is re-rendering.
+//
+// ```ts
+// export function MyComponent(props): JSX.Element {
+//     useWhyDidIRender('MyComponent', props)
+//     return <div>My Component</div>
+// }
+// ```
 export function useWhyDidIRender(name: string, props: Record<string, any>): void {
     const oldProps = useRef<Record<string, any>>()
```

Comment:
or literally replace this as a scene level thing that adds react scan when the flag is on